Sources/KlaviyoSwift/NetworkSession.swift

```diff
@@ -30,18 +30,36 @@ func createEmphemeralSession(protocolClasses: [AnyClass] = URLProtocolOverrides.
     return URLSession.init(configuration: configuration)
 }
 
+
 struct NetworkSession {
-    var data: (URLRequest) async throws -> (Data, URLResponse)
-    
-    static let production = {
-        let session = createEmphemeralSession()
-        return NetworkSession(data: { request async throws in
-            session.configuration.protocolClasses = URLProtocolOverrides.protocolClasses
-            return try await session.data(for: request)
-        })
-    }()
+        var data: (URLRequest) async throws -> (Data, URLResponse)
+        
+        static let production = { () -> NetworkSession in
+            let session = createEmphemeralSession()
+            
+            return NetworkSession(data: { request async throws -> (Data, URLResponse) in
+       
+                session.configuration.protocolClasses = URLProtocolOverrides.protocolClasses
+                if #available(iOS 15, *) {
+                    return try await session.data(for: request)
```

Comment:
Sadly this API is not available until iOS 15 so I fallback to my own version of it.